crates/db_schema/src/source/community.rs

```diff
@@ -68,7 +68,7 @@ pub struct CommunityForm {
   pub actor_id: Option<DbUrl>,
   pub local: Option<bool>,
   pub private_key: Option<Option<String>>,
-  pub public_key: String,
+  pub public_key: Option<String>,
```

Comment:
The main fix.

Reply:
This what I don't understand tho, public_key is always required, correct? So why make it optional? 

It seems to me the main problem is:

- Search for where PersonForm::default() and CommunityForm::default() are being used, and make sure the pubkey is always being set. Otherwise the String default of "" might get used. This might have just been a temporary bug in our API or apub code that no longer exists, but double check anyway.
- You correctly did a code migration to regenerate the missing ones.

Reply:
This is `CommunityForm`, not `Community`. The latter has public_key as a plain string, but for the form there is no reason to make it mandatory. You dont want to be forced to load and then write the public_key when you just want to update the description or mark community as hidden. I could change existing places where CommunityForm::default() etc are used, but I guarantee that the same problem would be introduced again in the future, and we wouldnt catch it in code review. Using an option here makes it impossible to overwrite the public_key by accident.